src/machine/asm/mod.rs

```diff
@@ -609,11 +690,26 @@ extern "C" {
 
 pub struct AsmMachine {
     pub machine: DefaultMachine<Box<AsmCoreMachine>>,
+    pub memory: Vec<u8>,
```

Comment:
I wasn't sure here, we are already dealing with raw pointers pretty much everywhere in this file, why not use `alloc::alloc` to allocate the memory? What's the benefit using `Vec` here? Yes we are saving us an impl of Drop but I wasn't sure if that's truly a benefit.

Reply:
Either vec or alloc will work, but maybe alloc is better.